web/packages/teleport/src/Discover/Database/DeployService/AutoDeploy/SelectSecurityGroups.tsx

```diff
@@ -268,18 +270,21 @@ function getTrustedSecurityGroups(
   securityGroups: SecurityGroup[],
   db: AwsRdsDatabase
 ): Set<string> {
-  const securityGroupsById = new Map(
-    securityGroups.map(group => [group.id, group])
-  );
+  const trustedGroups = new Set<string>();
+  if (!db) {
+    return trustedGroups;
+  }
 
   const dbPort = getPort(db);
-  const trustedGroups = new Set<string>();
-  db.securityGroups.forEach(groupId => {
+  const securityGroupsById = new Map(
+    securityGroups?.map(group => [group.id, group])
+  );
+  db?.securityGroups?.forEach(groupId => {
```

Comment:
`db?` is not necssary since we checked in at line 274

we also need to update the function parameter types to communicate it can be undefined

can we check for all undefines at line 274? `!db || !securityGroups`

Reply:
updated